pkg/api/v1/defaults.go

```diff
@@ -389,3 +389,32 @@ func SetDefaults_HostPathVolumeSource(obj *v1.HostPathVolumeSource) {
 		obj.Type = &typeVol
 	}
 }
+
+func SetDefaults_SecurityContext(sc *v1.SecurityContext) {
+	pt := true
+	if sc == nil {
+		sc = &v1.SecurityContext{
+			AllowPrivilegeEscalation: &pt,
+		}
+		return
+	}
+
+	// handle the case for AllowPrivilegeEscalation when the container is privileged
+	if sc.Privileged != nil && *sc.Privileged {
+		sc.AllowPrivilegeEscalation = &pt
+	}
+
+	// handle the case for AllowPrivilegeEscalation when we are adding CAP_SYS_ADMIN
+	if sc.Capabilities != nil {
```

Comment:
This `if` can be replaced by `else if` to prevent inspecting capabilities when we've already set `AllowPrivilegeEscalation` to `true` in the previous branch.